superset-frontend/src/components/Datasource/DatasourceEditor.jsx

```diff
@@ -698,6 +706,23 @@ class DatasourceEditor extends PureComponent {
     this.validate(this.onChange);
   }
 
+  async onQueryRun() {
+    this.props.runQuery({
+      client_id: this.props.clientId, // How this is generated?
```

Comment:
```suggestion
      client_id: this.props.clientId,
```

It's generated here: https://github.com/apache/superset/blob/4ed05f4ff16805b1253a902172a53e973cf41f1e/superset/utils/core.py#L1384

Reply:
solved